backend/src/sms/services/sms-callback.service.ts

```diff
@@ -94,7 +114,81 @@ const parseEvent = async (req: Request): Promise<void> => {
     )
   }
 }
+
+// for future context: we only care about when the twilio callback is a finalized status
+// we don't want to update the status pre-emptively until we receive the finalized status
+// initial state --> UNSENT
+const parseTransactionalEvent = async (req: Request): Promise<void> => {
+  const {
+    MessageStatus: twilioMessageStatus,
+    ErrorCode: twilioErrorCode,
+    MessageSid: messageId,
+  } = req.body
+  if (FINALIZED_STATUS.indexOf(twilioMessageStatus as string) === -1) {
+    return
+  }
+  logger.info({
+    message: 'Updating message in sms_messages_transactional',
+    messageId,
+    action: 'parseTransactionalEvent',
+  })
+
+  if (twilioErrorCode) {
+    await SmsMessageTransactional.update(
+      {
+        errorCode: twilioErrorCode,
+        status: TransactionalSmsMessageStatus.Error,
+      } as SmsMessageTransactional,
+      {
+        where: {
+          messageId: messageId,
+        },
+      }
+    )
+  } else {
+    // longer messages are delivered in multiple segments
+    // each segment has a separate delivery status
+    // Update the message as successful only if there does not exist previous failed status
+    // update only if sent or delivered
+    // map here because twilio does not explicitly send "error"
+    const mappedStatus = mapTwilioMessageStatus(twilioMessageStatus)
+    if (
+      mappedStatus === TransactionalSmsMessageStatus.Sent ||
+      mappedStatus === TransactionalSmsMessageStatus.Delivered
+    ) {
+      await SmsMessageTransactional.update(
+        {
+          updatedAt: new Date(),
+          status: mappedStatus,
+        } as SmsMessageTransactional,
+        {
+          where: {
+            messageId: messageId,
+            errorCode: { [Op.eq]: null },
+          },
+        }
+      )
+    }
+  }
+}
+
+const mapTwilioMessageStatus = (
+  status: string
+): TransactionalSmsMessageStatus => {
+  switch (status) {
+    case 'sent':
+      return TransactionalSmsMessageStatus.Sent
+    case 'delivered':
+      return TransactionalSmsMessageStatus.Delivered
+    case 'accepted':
+      return TransactionalSmsMessageStatus.Accepted
+    default:
+      return TransactionalSmsMessageStatus.Unsent
+  }
+}
 export const SmsCallbackService = {
   isAuthenticated,
```

Comment:
**[suggestion for future PR]** should we rename the campaign callback methods to specify they're for campaigns? would be clearer for future maintenance too imo